pkg/kedge/grpc/director/director.go

```diff
@@ -17,18 +19,30 @@ import (
 )
 
 // New builds a StreamDirector based off a backend pool and a router.
-func New(pool backendpool.Pool, router router.Router) proxy.StreamDirector {
+func New(pool backendpool.Pool, adhocRouter common.Addresser, grpcRouter router.Router) proxy.StreamDirector {
 	return func(ctx context.Context, fullMethodName string) (context.Context, *grpc.ClientConn, error) {
-		beName, err := router.Route(ctx, fullMethodName)
+		beName, err := grpcRouter.Route(ctx, fullMethodName)
 		if err != nil {
+			if err == router.ErrRouteNotFound {
+				fmt.Printf("director.go:27 debugging %#v\n", metautils.ExtractIncoming(ctx).Get(":authority"))
+				ipPort, err := adhocRouter.Address(metautils.ExtractIncoming(ctx).Get(":authority"))
+				if err != nil {
+					return ctx, nil, err
+				}
+				// I don't know which other dial options to include here
```

Comment:
you can inspire from: `kedge/grpc/backendpool/backend.go:125`

```
func BuildClientConn(cnf *pb.Backend, target string, resolver naming.Resolver) (*grpc.ClientConn, error) {
	var opts []grpc.DialOption
	opts = append(opts, chooseDialFuncOpt(cnf))
	opts = append(opts, chooseSecurityOpt(cnf))
	opts = append(opts, grpc.WithCodec(proxy.Codec())) // needed for the director to function at all.
	opts = append(opts, chooseInterceptors(cnf)...)
	opts = append(opts, grpc.WithBalancer(&addrTagBalancer{chooseBalancerPolicy(cnf, resolver)}))
	return grpc.Dial(target, opts...)
}
```
I think yours are fine but yu need also: `grpc.WithCodec(proxy.Codec()))` that might solve your issue